src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -298,5 +298,17 @@ public bool EqualsByName(CSharpType? other)
 
             return true;
         }
+
+        internal CSharpType WithArguments(IReadOnlyList<CSharpType> arguments)
```

Comment:
I think this should be `public`, and also could we name it as `MakeGenericType` to mimic the same behavior in `System.Type`? I think this method serves the same purpose to replace the open generic parameter to a concrete type, such as create `IList<int>` out of `IList<T>`